retry/get_body_go18.go

```diff
@@ -16,8 +16,12 @@ func getBody(r *http.Request) func() (io.ReadCloser, error) {
 	} else if r.Body != nil {
 		// Optimise for io.ReadSeeker (e.g file readers) for uploading large files.
 		if rs, ok := r.Body.(io.ReadSeeker); ok {
+			currentPosition, err := rs.Seek(0, io.SeekCurrent)
+			if err != nil {
+				return nil
```

Comment:
Return a func that returns the error similar to what happens below:

```suggestion
				return func() (io.ReadCloser, error) {
				    return nil, err
			        }
```